test/Feature/ResourceArrays/multi-dim-array-subset.test

```diff
@@ -0,0 +1,101 @@
+#--- source.hlsl
+
+// This test verified handling of subsets of multi-dimensional resource
+// arrays used in a function argument.
+
+RWBuffer<float> In[4][2] : register(u0);
+RWBuffer<float> Out : register(u0, space1);
+RWBuffer<float> Out2 : register(u1, space1);
+
+float foo(RWBuffer<float> A[2], uint Index) {
+  return A[0][Index] + A[1][Index];
+}
+
+[numthreads(4,1,1)]
+void main(uint GI : SV_GroupIndex) {
+  // the content of Out1 and Out2 should be identical
+  Out1[GI] = foo(In[GI], GI);
+  Out2[GI] = In[GI][0][GI] + In[GI][1][GI];
+}
+
+//--- pipeline.yaml
+---
+Shaders:
+  - Stage: Compute
+    Entry: main
+    DispatchSize: [1, 1, 1]
+Buffers:
+  - Name: BufIn
+    Format: Float32
+    ArraySize: 8
+    Data:
+      - [ 1.0, 2.0, 3.0, 4.0 ]
+      - [ 0.1, 0.2, 0.3, 0.4 ]
+      - [ 5.0, 6.0, 7.0, 8.0 ]
+      - [ 0.5, 0.6, 0.7, 0.8 ]
+      - [ 10.0, 11.0, 12.0, 13.0 ]
+      - [ 0.10, 0.11, 0.12, 0.13 ]
+      - [ 14.0, 15.0, 16.0, 17.0 ]
+      - [ 0.14, 0.15, 0.16, 0.17 ]
+
+  - Name: BufOut1
+    Format: Float32
+    ZeroInitSize: 16
+
+  - Name: BufOut2
+    Format: Float32
+    ZeroInitSize: 16
+
+  - Name: ExpectedOuts
+    Format: Float32
+    Data: [ 1.1, 6.6, 12.12, 17.17 ]
```

Comment:
Can somebody please double check my math, if the expected results are correct?

DXC output is `[ 2, 12, 13.2, 18.7 ]`. I can't figure out how could it come up with that, given that each `BufOut1` value should be a sum of 2 values from `BufIn`.

Reply:
The math looks correct to me. Even tried asking co-pilot to confirm that.
Is it possible there is an issue because In[4][2] is assigned to register(u0), and then Out is register(u0, space1)?

Reply:
Please disregard - this was a user error :) This is working correctly on DXC. I probably had the test source in a bad state so it did not compile, and the test ran and compared some older code version with new pipeline definition.